src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.List;
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Bulldozer.class)) {
+            return (List<Machine>) new BulldozerProducer().get();
+        }
+        if (type.equals(Excavator.class)) {
+            return (List<Machine>) new ExcavatorProducer().get();
+        }
+        if (type.equals(Truck.class)) {
+            return (List<Machine>) new TruckProducer().get();
+        }
+        return null;
```

Comment:
Sofia will say you that you return an empty list using the method of class Collections

Reply:
ok. Changed